calour/filtering.py

```diff
@@ -245,76 +235,52 @@ def filter_by_data(exp: Experiment, predicate, axis=1, field=None, negate=False,
     kwargs : dict
         keyword argument passing to predicate function.
 
-    Returns
-    -------
-    Experiment
-        the filtered object
-
     See Also
     --------
     filter_mean_abundance
-    filter_abundance
+    filter_sum_abundance
     filter_prevalence
 
     '''
+    if exp.normalized <= 0:
+        logger.warning('Do you forget to normalize your data? It is required before running this function')
+    logger.debug('filter_by_data using function %r' % predicate)
+
     if axis == 0:
+        metadata = exp.feature_metadata
         # transpose it so all the following operations are performed on column
-        x = exp.feature_metadata
         data = exp.data.T
     elif axis == 1:
-        x = exp.sample_metadata
+        metadata = exp.sample_metadata
         data = exp.data
-    else:
-        raise ValueError('unknown axis %s' % axis)
```

Comment:
again why delete this error?

Reply:
duplicate. it is raised in _convert_axis_name().